backends/platform/atari/atari.cpp

```diff
@@ -0,0 +1,643 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <time.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include <mint/cookie.h>
+#include <mint/falcon.h>
+#include <mint/osbind.h>
+
+// We use some stdio.h functionality here thus we need to allow some
+// symbols. Alternatively, we could simply allow everything by defining
+// FORBIDDEN_SYMBOL_ALLOW_ALL
+#define FORBIDDEN_SYMBOL_EXCEPTION_FILE
+#define FORBIDDEN_SYMBOL_EXCEPTION_stdout
+#define FORBIDDEN_SYMBOL_EXCEPTION_stderr
+#define FORBIDDEN_SYMBOL_EXCEPTION_fputs
+#define FORBIDDEN_SYMBOL_EXCEPTION_exit
+#define FORBIDDEN_SYMBOL_EXCEPTION_time_h
+
+#include "common/scummsys.h"
+
+#if defined(ATARI)
+#include "backends/graphics/atari/atari-graphics-asm.h"
+#include "backends/keymapper/hardware-input.h"
+#include "backends/modular-backend.h"
+#include "backends/mutex/null/null-mutex.h"
+#include "base/main.h"
+
+#include "backends/saves/default/default-saves.h"
+#include "backends/timer/default/default-timer.h"
+#include "backends/events/default/default-events.h"
+#include "backends/mixer/atari/atari-mixer.h"
+#include "backends/graphics/atari/atari-graphics.h"
+#include "backends/graphics/atari/atari-graphics-supervidel.h"
+#include "backends/graphics/atari/atari-graphics-videl.h"
+#include "common/hashmap.h"
+#include "gui/debugger.h"
+
+/*
+ * Include header files needed for the getFilesystemFactory() method.
+ */
+#include "backends/fs/posix/posix-fs-factory.h"
+
+class OSystem_Atari : public ModularMixerBackend, public ModularGraphicsBackend, Common::EventSource {
```

Comment:
This should go in a `osystem_atari.h` file with a `osystem_atari.cpp` file for the implementation.

Reply:
I've just followed the example of null system. Also I'd argue that if `osystem_atari.h` is never included anywhere, why create it in the first place.

Reply:
In the end, I had to do it for the AtariEventSource separation, so - done.